hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/PendingDeletionEndpoint.java

```diff
@@ -0,0 +1,117 @@
+import java.util.concurrent.CompletableFuture;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.hdds.scm.protocol.StorageContainerLocationProtocol;
+import org.apache.hadoop.ozone.recon.api.types.DataNodeMetricsServiceResponse;
+import org.apache.hadoop.ozone.recon.api.types.ScmPendingDeletion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * REST API endpoint that provides metrics and information related to
+ * pending deletions. It responds to requests on the "/pendingDeletion" path
+ * and produces application/json responses.
+ */
+@Path("/pendingDeletion")
+@Produces("application/json")
+@AdminOnly
+public class PendingDeletionEndpoint {
+  private static final Logger LOG = LoggerFactory.getLogger(PendingDeletionEndpoint.class);
+  private final ReconGlobalMetricsService reconGlobalMetricsService;
+  private final DataNodeMetricsService dataNodeMetricsService;
+  private final StorageContainerLocationProtocol scmClient;
+
+  @Inject
+  public PendingDeletionEndpoint(ReconGlobalMetricsService reconGlobalMetricsService,
+                                 DataNodeMetricsService dataNodeMetricsService,
+                                 StorageContainerLocationProtocol scmClient) {
+    this.reconGlobalMetricsService = reconGlobalMetricsService;
+    this.dataNodeMetricsService = dataNodeMetricsService;
+    this.scmClient = scmClient;
+  }
+
+  @GET
+  public Response getPendingDeletionByComponent(@QueryParam("component") String component) {
+    if (component == null || component.isEmpty()) {
+      return Response.status(Response.Status.BAD_REQUEST)
+          .entity("component query parameter is required").build();
+    }
+    final String normalizedComponent = component.trim().toLowerCase();
+    switch (normalizedComponent) {
+    case "dn":
+      return handleDataNodeMetrics();
+    case "scm":
+      return handleScmPendingDeletion();
+    case "om":
+      return handleOmPendingDeletion();
+    default:
+      return Response.status(Response.Status.BAD_REQUEST)
+          .entity("component query parameter must be one of dn, scm, om").build();
+    }
+  }
+
+  private Response handleDataNodeMetrics() {
+    DataNodeMetricsService.MetricCollectionStatus status = dataNodeMetricsService.getTaskStatus();
+    if (status != DataNodeMetricsService.MetricCollectionStatus.IN_PROGRESS) {
+      CompletableFuture.runAsync(dataNodeMetricsService::startTask);
+    }
+    DataNodeMetricsServiceResponse response = dataNodeMetricsService.getCollectedMetrics();
+    if (response.getStatus() != DataNodeMetricsService.MetricCollectionStatus.SUCCEEDED) {
+      return Response.accepted(DataNodeMetricsServiceResponse.newBuilder()
+          .setStatus(DataNodeMetricsService.MetricCollectionStatus.IN_PROGRESS).build())
```

Comment:
Returns HTTP 202 (Accepted) for both NOT_STARTED and IN_PROGRESS states, but always reports status as IN_PROGRESS in response body. How about below?

```suggestion
          DataNodeMetricsServiceResponse actualResponse = dataNodeMetricsService.getCollectedMetrics();
if (actualResponse.getStatus() != DataNodeMetricsService.MetricCollectionStatus.SUCCEEDED) {
  return Response.accepted(actualResponse).build();  // Return actual status, not hardcoded IN_PROGRESS
}
```